### ISSUE TITLE
There are always some missing checks.why？

### ISSUE THREAD

**Issue by baolinhu** [AlexeyAB/darknet#1014]
There are always missing tests, which are usually easy to detect.do you know why？thanks.
![image](https://user-images.githubusercontent.com/16516384/41189827-7aafddd2-6c06-11e8-8bc3-23d755d643fa.png)
![image](https://user-images.githubusercontent.com/16516384/41189833-84f1fd8e-6c06-11e8-8ff6-68343a0c9215.png)

**Comment by AlexeyAB** [comment]
Many reasons. Need more info.

**Comment by baolinhu** [comment]
I only detect car.about 9000 images,learning_rate=0.0001,max_batches=3000,steps=1500,2400,scales=.1,.1.
The loss was great at the end of the training,about 1.
Is it a question of learning rate？
thanks.

**Comment by baolinhu** [comment]
mAP=87.64%.It‘’s too big.Maybe there is something wrong.
![image](https://user-images.githubusercontent.com/16516384/41192688-401deb9a-6c34-11e8-8c9c-7f612f0d3e14.png)
model is darknet53.
width=height=512
I used pre-trained weights.
sorry，my lr=0.001.I write wrong.

**Comment by baolinhu** [comment]
I use default thresh 0.24.
If I use smaller thresh such as 0.15,it may be detected.But there will be some other mistakes.

**Comment by AlexeyAB** [comment]
* darknet53 is classifier, isn't detector, it can't detect objects

* Set steps `steps=8000,9000` and train from the begining 10 000 iterations

**Comment by baolinhu** [comment]
what you mean model ? default model is darknet53.The cfg file I changed from yolov3-voc.cfg.
so many iterations for only one class?
I will try it. thanks.

**Comment by AlexeyAB** [comment]
* Perhaps not all objects are labeled in your dataset: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> check that each object are mandatory labeled in your dataset - no one object in your data set should not be without label. In the most training issues - there are wrong labels in your dataset (got labels by using some conversion script, marked with a third-party tool, ...). Always check your dataset by using: https://github.com/AlexeyAB/Yolo_mark


* Also what can you get by this command?
    `./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 512 -height 512`

**Comment by baolinhu** [comment]
> * I use an open competition dataset, and there are some problems with the labels. little cars are very obscured or at the boundaries of images. It is difficult to decide whether to be labeled or not.
> * In normal , how big is the final loss of training?
![image](https://user-images.githubusercontent.com/16516384/41209172-4aafbb28-6d5c-11e8-98c3-4e194850a53c.png)

**Comment by baolinhu** [comment]
- This result of "calc_anchors" is very strange. Should I modify the default "anchors" ratio?

**Comment by AlexeyAB** [comment]
@baolinhu Update your code from GitHub and do calc_anchors again.

**Comment by baolinhu** [comment]
- I updated the code to recalculate.It is normal.Maybe I need change anchors ratios. Is that right？
![image](https://user-images.githubusercontent.com/16516384/41240559-66d450ea-6dcd-11e8-95db-43503f51e126.png)
- I think your code has a little bug.“replace_image_to_label”，This function will go wrong. I used the previous version to replace JPEGIages and labels.You can see in above picture.
![image](https://user-images.githubusercontent.com/16516384/41240637-975ac3d4-6dcd-11e8-975b-5e018cc12d3d.png)

**Comment by AlexeyAB** [comment]
@baolinhu 

> I updated the code to recalculate.It is normal.Maybe I need change anchors ratios. Is that right？

Yes, it is normal. What do you mean by change anchors ratios?


> I think your code has a little bug.“replace_image_to_label”，This function will go wrong. I used the previous version to replace JPEGIages and labels.You can see in above picture.

* What do you mean? 
* What dataset do you use?
* What directories do you use for Images and Labels?

Current code:  https://github.com/AlexeyAB/darknet/blob/eff487ba3626a39e135d13929117e04bc4cf5823/src/utils.c#L208-L229

**Comment by baolinhu** [comment]
-  1.I updated the code yesterday.I have seen the "replace_image_to_label" function.But it will go wrong.Anyway,do you think my calculation of "anchors" is normal or right?
-  2.The dataset I used was made by myself, and the format is the same as VOC.
-  3.You can see the directory structure in the following picture.
![image](https://user-images.githubusercontent.com/16516384/41265651-f02fa120-6e25-11e8-996f-3fd17a14f0da.png)
-  4.If I use this function(which means "replace_image_to_label"), it will go wrong, and I can't find TXT under "labels".
![image](https://user-images.githubusercontent.com/16516384/41265667-035ad01c-6e26-11e8-925b-9a0402a9bc5f.png)

**Comment by AlexeyAB** [comment]
1. Yes
2. Did you download Pascal VOC dataset from this site? https://pjreddie.com/projects/pascal-voc-dataset-mirror/
3. Did you rename this directory `VOCdevkit\VOC2012\JPEGImages` to this `data\DC_VOC\JPEGImages`?

**Comment by baolinhu** [comment]
-  1. thanks.I will change the default anchors ratios and train again.
-  2. I used the VOC dataset in the past.
-  3. No.I only used Annotations、ImageSets and JPEGImages without VOC2007 or VOC2012.I think previous version is more flexible.But both are ok.Thanks.

**Comment by AlexeyAB** [comment]
3. Some people used `/JPEGImages` folder for its own datasets (not PascalVOC), with images and labels in the same directory, and function `find_replace(output_path, "JPEGImages", "labels", output_path);` broke training.

**Comment by baolinhu** [comment]
ok,I know.Thanks.
I first trained a weights. Then I want to use this weights as a pretrained model and then use another dataset continue training.
- 1. Is that all of the layers trained?
- 2. How to fix some layers and only train the last some layers(such as after 81 layers)?

**Comment by baolinhu** [comment]
The original picture is 1069*500.
- 1.Can the input h and w be not equal? As long as it can be divisible by 32?
- 2.Is the ratio as close as possible to the original picture correct?
- 3.What size training should I use?

**Comment by AlexeyAB** [comment]
1. You can use any size of images. You can use any size of network that is divisible by 32, for example width=1056 height=480

2. Yes. It is advisable to do so the ratio as close as possible to the original picture

3. Size of what?

**Comment by baolinhu** [comment]
I used my own training weights to pretrain, but the result was very poor. Maybe my way is wrong. I'll try again tomorrow. I should reduce the  lr and smaller max_batch training, right?Thanks a lot.

**Comment by baolinhu** [comment]
I want to increase the size of pictures.Because of the limited capacity of my graphics card.I have two questions.
- 1.What is the difference between increasing subdivision and reducing batch?
- 2.Will increasing subdivision reduce the accuracy rate?

**Comment by AlexeyAB** [comment]
1. increasing subdivision only decreases mini-batch, but reducing batch decreases number of mini-batches for 1 iteration. So you should set batch=64 or 128, not lower.

2. Yes, but only slightly

**Comment by baolinhu** [comment]
- The size of all my pictures is the same size. Can I turn off the multi-scale training and let random=0 affect the accuracy?
- Because I use 1056*480, its scale becomes 1440*640, and my computer will be oom. The training stopped. I have only 2 Titan graphics cards

**Comment by AlexeyAB** [comment]
Yes, you can use random=0, but it will decrease accuracy.

**Comment by baolinhu** [comment]
@AlexeyAB 
![image](https://user-images.githubusercontent.com/16516384/41454938-d8d1819e-70ad-11e8-982b-6bee7887bca8.png)
- My average IoU = 80.58%, mAP= 88.18% is not high enough. What do you suggest to improve IoU and mAP? Thank you.
- I'm using 1056*480's image now. I only detect one class car, and all kinds of cars (car, bus, truck, etc.) are labeled car. The following are my parameters:
`Random=0,
Learning_rate=0.001
Burn_in=1000
Max_batches = 8000
Policy=steps
Steps=60007000
Scales=.1,.1`

**Comment by AlexeyAB** [comment]
* Do you use images 1056x480 or network resolution 1056x480?
* What width= and height= in the cfg-file?
* Did you re-calculate anchors for new network resolution if you use width=1056 height=480?
    `./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 512 -height 512`
* How many images do you use?

**Comment by baolinhu** [comment]
@AlexeyAB 
- 1.Some pictures have a lot of omission which are easy to be detected and I used a very low thresh about 0.12~0.15, but most of them are not.
![image](https://user-images.githubusercontent.com/16516384/41495381-7bc278f4-7158-11e8-9254-324ef01f5209.png)
![image](https://user-images.githubusercontent.com/16516384/41495400-e4294bb6-7158-11e8-98aa-4dd4609a37d5.png)
- 2.The boxes positioning of some vehicles at the bottom of the image is very inaccurate, too big.
![image](https://user-images.githubusercontent.com/16516384/41495411-3e11be56-7159-11e8-8d63-4739981d0a97.png)

- 3.My max_batch=8000, is my training inadequate? How do you make the box more accurate?
- 4.Should I increase the proportion of boxes's loss? I didn't find a loss ratio parameter like yolov2 in cfg file.   `object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1`
Thank you very much. The answer is always so timely.

**Comment by AlexeyAB** [comment]
> 4.About **11000** for only one class "car",and train:val = 9:1..But the accuracy of 10000 of the GT boxes is 85% (which means that manual handwriting is not so accurate。Some GT boxes may not be so accurate, maybe some boxes are large or little leaks . But it can be used). The accuracy of the remaining 1000 images' GT boxes is 99%.

1. Try to train about 15 000 - 20 000 iterations.

2. Try to set `random=1`

3. Also it seems that not all objects are labeled in your training dataset.


> 4.Should I increase the proportion of boxes's loss? I didn't find a loss ratio parameter like yolov2 in cfg file. object_scale=5 noobject_scale=1 class_scale=1 coord_scale=1
> Thank you very much. The answer is always so timely.

There isn't such parameters in the Yolo v3 [yolo]-layer.

**Comment by baolinhu** [comment]
@AlexeyAB

> Try to train about 15 000 - 20 000 iterations.

Try to set random=1

- 1. I trained 20000 times, and the effect was worse.
- 2.Do you know why 'Couldn't open file: /dlw/hbl/code/github/darknet/data/voc/train.txt', when I use the official code.I didn't find the differences with the two codes in reading files.
- 3.I find that yolov3 is not very accurate in locating large objects. Is there any way to improve it?

Thanks.

**Comment by AlexeyAB** [comment]
> 3.I find that yolov3 is not very accurate in locating large objects. Is there any way to improve it?

Just use the correct anchors, and correct dataset.
